utils/generate_grammar_tables.py

```diff
@@ -34,6 +36,17 @@
 SPV_AMD_shader_trinary_minmax
 """
 
+def write_to_file(data, filename):
+    temp_file = tempfile.NamedTemporaryFile(mode='w', dir=os.path.dirname(filename), delete=False)
+    print(data,file=temp_file)
+    temp_file.flush()
+    temp_file.close()
+    try:
+      os.rename(temp_file.name, filename)
+    except OSError:
+      # Can happen on Windows when the file arleady exists.  We can just delete
```

Comment:
I think this will have problems for incremental build when the grammar changes? The outdated file will always be in the binary directory preventing us from writing to it. It seems we should compare the contents and do force replacement if they differ?

Reply:
Right.  Fine for Unix systems, but not for Windows.